File: Bookcase core/src/test/java/cz/vhromada/book/converter/BookConverterTest.java
package cz.vhromada.book.converter;

import static org.assertj.core.api.Assertions.assertThat;

import cz.vhromada.book.CoreTestConfiguration;
import cz.vhromada.book.entity.Book;
import cz.vhromada.book.utils.BookUtils;
import cz.vhromada.converter.Converter;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit.jupiter.SpringExtension;

/**
 * A class represents test for converter between {@link cz.vhromada.book.domain.Book} and {@link Book}.
 *
 * @author Vladimir Hromada
 */
@ExtendWith(SpringExtension.class)
@ContextConfiguration(classes = CoreTestConfiguration.class)
class BookConverterTest {

    /**
     * Instance of {@link Converter}
     */
    @Autowired
    private Converter converter;

    /**
     * Test method for {@link Converter#convert(Object, Class)} from domain to entity.
     */
    @Test
    void convertBookDomain() {
        final cz.vhromada.book.domain.Book bookDomain = BookUtils.newBookDomain(1);
        final Book book = converter.convert(bookDomain, Book.class);

        BookUtils.assertBookDeepEquals(book, bookDomain);
    }

    /**
     * Test method for {@link Converter#convert(Object, Class)} from domain to entity with null book.
     */
    @Test
    void convertBookDomain_NullBook() {
        assertThat(converter.convert(null, Book.class)).isNull();
    }

    /**
     * Test method for {@link Converter#convert(Object, Class)} from entity to domain.
     */
    @Test
    void convertBook() {
        final Book book = BookUtils.newBook(1);
        final cz.vhromada.book.domain.Book bookDomain = converter.convert(book, cz.vhromada.book.domain.Book.class);

        BookUtils.assertBookDeepEquals(book, bookDomain);
    }

    /**
     * Test method for {@link Converter#convert(Object, Class)} from entity to domain with null book.
     */
    @Test
    void convertBook_NullBook() {
        assertThat(converter.convert(null, cz.vhromada.book.domain.Book.class)).isNull();
    }

}

File: Bookcase core/src/test/kotlin/cz/vhromada/book/converter/BookConverterTest.kt
package cz.vhromada.book.converter

import cz.vhromada.book.CoreTestConfiguration
import cz.vhromada.book.domain.Book
import cz.vhromada.book.utils.BookUtils
import cz.vhromada.converter.Converter
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * A class represents test for class [BookConverter] and [DomainBookConverter].
 *
 * @author Vladimir Hromada
 */
@ExtendWith(SpringExtension::class)
@ContextConfiguration(classes = [CoreTestConfiguration::class])
class BookConverterTest {

    /**
     * Instance of [Converter]
     */
    @Autowired
    private val converter: Converter? = null

    /**
     * Checks autowired fields.
     */
    @BeforeEach
    fun setUp() {
        assertThat(converter).isNotNull
    }

    /**
     * Test method for [BookConverter.convert].
     */
    @Test
    fun convert_Entity() {
        val book = BookUtils.newBook(1)
        val bookDomain = converter!!.convert(book, Book::class.java)

        BookUtils.assertBookDeepEquals(bookDomain, book)
    }

    /**
     * Test method for [DomainBookConverter.convert].
     */
    @Test
    fun convert_Domain() {
        val bookDomain = BookUtils.newBookDomain(1)
        val book = converter!!.convert(bookDomain, cz.vhromada.book.entity.Book::class.java)

        BookUtils.assertBookDeepEquals(bookDomain, book)
    }

}
